tests/textlab_api/textlab_api_test.py

```diff
@@ -46,7 +44,256 @@ def update_page_content(
+    # Enable Textlab in the test region
+    Region.objects.filter(slug="augsburg").update(hix_enabled=True)
+
+    response = create_page(
+        admin_client,
+        "Test title: test_ignore_hix_on_page_create",
+        "test_ignore_hix_on_page_create: Neuer Inhalt",
+        hix_ignore=True,
+    )
+
+    assert response.status_code == 302
+
+    # Check the saved HIX score
+    page_translation = Page.objects.latest("created_date").get_translation("de")
+
+    assert page_translation.hix_score is None
+    assert page_translation.hix_feedback is None
+    assert mock_server.requests_counter == 0
+
+
+@pytest.mark.django_db
+def test_hix_disabled_on_region_on_page_create(
+    load_test_data: None,
+    admin_client: Client,
+    settings: SettingsWrapper,
+    mock_server: MockServer,
+) -> None:
+    """
+    Check that the HIX score is not requested on page create when hix is disabled for region
+    No request to TextLab API is expected.
+
+    :param load_test_data: The fixture providing the test data (see :meth:`~tests.conftest.load_test_data`)
+    :param admin_client: The fixture providing the http client
+    :param settings: The fixture providing the django settings
+    :param mock_server: The fixture providing the dummy http server
+    """
+    # Setup a mocked Textlab API server with dummy responses
+    mock_server.configure("/user/login", 200, {"token": "dummy"})
+    mock_server.configure("/benchmark/420", 200, {"formulaHix": 15.48920568})
+
+    # Redirect call aimed at the Textlab API to the fake server
+    settings.TEXTLAB_API_URL = f"http://localhost:{mock_server.port}"
+
+    # Enable Textlab in the test region
+    Region.objects.filter(slug="augsburg").update(hix_enabled=False)
+
+    response = create_page(
+        admin_client,
+        "Test title: test_hix_disabled_on_region_on_page_create",
+        "test_hix_disabled_on_region_on_page_create: Neuer Inhalt",
+    )
+
+    assert response.status_code == 302
+
+    # Check the saved HIX score
+    page_translation = Page.objects.latest("created_date").get_translation("de")
+
+    assert page_translation.hix_score is None
+    assert page_translation.hix_feedback is None
+    assert mock_server.requests_counter == 0
+
+
+@pytest.mark.django_db
+def test_hix_response_400_on_page_create(
+    load_test_data: None,
+    admin_client: Client,
+    settings: SettingsWrapper,
+    mock_server: MockServer,
+) -> None:
+    """
+    Check that the HIX score is not requested on page create when hix is enabled but returns status 400
+
+    :param load_test_data: The fixture providing the test data (see :meth:`~tests.conftest.load_test_data`)
+    :param admin_client: The fixture providing the http client
+    :param settings: The fixture providing the django settings
+    :param mock_server: The fixture providing the dummy http server
+    """
+    # Setup a mocked Textlab API server with dummy responses
+    mock_server.configure("/user/login", 200, {"token": "dummy"})
```

Comment:
Nice-to-have improvement:
We could return 400 to login request in this test.
Then we'll check the benchmark error in the test_hix_response_400_on_page_update() and the logic error here.
So the coverage will be a little better.

Reply:
Good point, fixed